lib/Transforms/IPO/PagerandoWrappers.cpp

```diff
@@ -149,8 +148,11 @@ static void replaceAddressTakenUse(Use *U, Function *F, Function *Wrapper,
 
 Function *PagerandoWrappers::createWrapper(Function &F,
                                            const std::vector<Use *> &AddressUses) {
-  auto Wrapper = Function::Create(F.getFunctionType(), F.getLinkage(),
-                                  F.getName() + WrapperSuffix, F.getParent());
+  std::string Name = F.getName();
+  F.setName(Name + (F.isVarArg() ? OrigVASuffix : OrigSuffix));
+
+  auto Wrapper = Function::Create(F.getFunctionType(), F.getLinkage(), Name,
```

Comment:
This is ok with me, but it does change semantics. Previously we used `$$wrap` for wrappers that are not externally visible but are address-taken, which could escape. Replacing the name entirely seems fine for this case too.

Reply:
Yes, I thought this is more consistent/less confusing, wrapper now always has the "clean" name and the original function always has the suffix.
Commit message:
```
If a wrapper is created it will now always take the name of the
original function even if not strictly necessary (i.e., for
functions which only need a wrapper because they have their
address taken).
The original function will be renamed to <name>$$orig
(<name>$$origva for vararg functions).
```